include/dlaf/common/index2d.h

```diff
@@ -224,9 +224,12 @@ Index2D<IndexT, Tag> computeCoordsRowMajor(std::ptrdiff_t linear_index,
                                            const Size2D<IndexT, Tag>& dims) noexcept {
   using dlaf::util::ptrdiff_t::mul;
 
-  DLAF_ASSERT_MODERATE(linear_index >= 0, "The linear index cannot be negative!", linear_index);
+  // `linear_index` is wrapped with `std::to_string` because uint8_t or int8_t are interpreted as chars,
```

Comment:
If you want to leave the comment, it's ok for me, the only thing is that I don't know how useful it can be. Just to highlight that these are not the only places where there is this issue, in the past I had to fix this in another place.

Reply:
Yeah, there are indeed more than one places where this occurs. I appreciate having it once as a reminder as it is non-trivial and may come-up again. On the other hand, I didn't want to add it everywhere as it's too much bloat. Should we just leave it as it is now?